src/arraystring.rs

```diff
@@ -588,11 +590,15 @@ impl<SIZE: Capacity> ArrayString<SIZE> {
     /// ```rust
     /// # use arraystring::prelude::*;
     /// # let _ = env_logger::try_init();
-    /// assert_eq!(ArrayString::<typenum::U32>::capacity(), 32);
+    /// assert_eq!(ArrayString::<32>::capacity(), 32);
     /// ```
     #[inline]
-    pub fn capacity() -> u8 {
-        SIZE::to_u8()
+    pub const fn capacity() -> u8 {
+        if N > u8::MAX as usize {
```

Comment:
only usize, char and bool on stable, on nightly u8 is available with extended types.

i'll soon make a PR with some cleanup to avoid redundancy so we can impl stuff easily, with a ArrayStringBase containing all the base operations, and ArrayStringExt that is impl ArrayStringExt for T where T: ArrayStringBase;
it should allow to do your hack without defining it on all functions.
however i propose to add a nightly feature where we can enable the proper way using a u8 and disable the hack.

Reply:
you could also do it like
```rust

trait AllowedRange {}
impl AllowedRange for ArrayString<7> {}

#[derive(Copy, Clone)]
pub struct ArrayString<const N: usize> where Self: AllowedRange { ... }

impl<const N: usize> ArrayString<N> where Self: AllowedRange { // all impl blocks
}
```

Reply:
That is nice! Didn't know you could require a trait defined for the struct in the struct definition.

I totally agree with allowing nightly to disable the hack, but until u8 can be accepted in the const generic definition we need one of the two.